src/pages/EditMonitor.vue

```diff
@@ -1157,6 +1157,23 @@ export default {
     },
 
     computed: {
+        defaultFriendlyName() {
+            if (this.monitor.hostname) {
+                return this.monitor.hostname;
+            }
+            if (this.monitor.url) {
+                try {
+                    const url = new URL(this.monitor.url);
+                    return url.hostname;
+                } catch (e) {
+                    // Handle invalid URL, maybe return a generic placeholder or empty string
+                    return this.$t("My Monitor");
```

Comment:
I think this is a better default here

```suggestion
                    return this.monitor.url.replace(/https?:\/\//, "")
```